enhancements/kube-apiserver/stability-point-to-point-network-check.md

```diff
@@ -82,6 +82,80 @@ What are the caveats to the implementation? What are some important details that
 didn't come across above. Go in to as much detail as necessary here. This might
 be a good place to talk about core concepts and how they relate.
 
+#### RelatedEndpoint Custom Resource
+
+```go
+package v1alpha1
+import 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+// RelatedEndpoint references a "interesting" endpoint for a given pod.
+type RelatedEndpoint struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata"`
+    
+    Spec RelatedEndpointSpec `json:"spec"`
+    Status RelatedEndpointStatus `json:"status"`
+}
+
+type RelatedEndpointSpec struct {
+    Pod string `json:"string"`
+    Endpoint RelatedEndpointSpecEndpoint `json:"endpoint"`
+}
+
+type RelatedEndpointSpecEndpoint struct {
+    // URL is the url of the end point
+    URL string `json:"url"`
+    // CACertFile, if specified, will be used to validate server certs
+    CACertFile string `json:"url,omitempty"`
```

Comment:
We can use insecure for these, right? We aren't doing anything secret.  Just trying to get a connection.

Reply:
We can. I wanted some comments as to if this was useful or not. It would be nice to log the successful response for example, but I'm not sure if that would be too out of scope.

Reply:
(^^^ comment assumed you were talking about all the crypto options, but yes, we can just use insecure w/ client cert/key to get a connection)